### PR TITLE
Made config inheritance recursive. Added working configs for weak_therapist and mixed_therapist

### PR DESCRIPTION
**Pull request by micahcarroll**
## Description
- Fixing autocopy and related SLURM scripts to work both on CHAI and CAIS clusters. Note that CAIS cluster will ignore the amount of memory you request (not sure yet if this will be an issue, hopefully not)
- Made config system be able to handle multi-step inheritance A <- B <- C
- Updated handling of API_KEYs to be read from `~/.env`. @carolius, @AdhyyanNarang, @weissercn I know you all have your keys in your `~/.bashrc` and they should not be there.
- Got working results for weak_therapist and mixed_therapist

## Type of change
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. 

- [x] I have run the local tests with `pytest --gpus=X`.
- [ ] I have run the experiment with `EI_test_up.yaml`
- [x] I have run full experiments with weak_therapist and mixed_therapist which work

### CODE REVIEW

File: influence_benchmark/config/accelerate_config.py [hunk 1]
```diff
@@ -24,8 +26,13 @@ def set_gpu_ids(self, gpu_ids: Optional[List[int]]):
             return
         # NOTE: Currently only support one GPU, maybe this is not what we want for KTO?
         max_gpus = 1
-        self.gpu_ids = gpu_ids[:max_gpus]
+        if len(gpu_ids) > max_gpus:
+            gpu_ids = np.random.choice(gpu_ids, max_gpus, replace=False).tolist()
```
Comment (carolius) [inline influence_benchmark/config/accelerate_config.py:30]: seems quite convoluted to do a random gpu? we could just do the first one?

File: influence_benchmark/config/experiment_configs/EI_test.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ final_reward: false
 num_gen_trajs_per_initial_state: 1
 top_n_trajs_per_initial_state: 1
 iterations: 1
-pm_length_penalty: null
+pm_length_penalty: 5.0e-4
```
Comment (carolius) [inline influence_benchmark/config/experiment_configs/EI_test.yaml:17]: I think this is quite a high length penalty? I usually do a factor 10 less. Not sure what the optimum is

File: influence_benchmark/config/experiment_configs/EI_mixed_therapist_1_step.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+parent_config_to_override: "EI_mixed_therapist.yaml"
+run_name: "EI-mixed-therapist-1-step"
+
+max_turns: 1
+num_envs_per_device: 30
```
Comment (carolius) [inline influence_benchmark/config/experiment_configs/EI_mixed_therapist_1_step.yaml:5]: will 30 fit on a A6000?

File: influence_benchmark/root.py [hunk 4]
```diff
@@ -1,5 +1,9 @@
 from pathlib import Path
 
+from dotenv import load_dotenv
```
Comment (carolius) [inline influence_benchmark/root.py:3]: does this mean that only code which at some point uses a path in this file will have access to api keys?

File: server_setup_scripts/setup_server.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
Comment (carolius) [inline server_setup_scripts/setup_server.sh:1]: why is there both [scripts/setup_server.sh] and [server_setup_scripts/setup_server.sh]?

File: influence_benchmark/config/accelerate_config.py (hunk 1 above)
Comment (micahcarroll) [reply influence_benchmark/config/accelerate_config.py:30]: Yeah i had a reason I wanted to do random, but turned out to not actually help much at all. Will revert

File: influence_benchmark/config/experiment_configs/EI_test.yaml (hunk 2 above)
Comment (micahcarroll) [reply influence_benchmark/config/experiment_configs/EI_test.yaml:17]: I don't care, this is just for testing that it doesn't crash and burn

Comment (micahcarroll) [reply influence_benchmark/config/experiment_configs/EI_test.yaml:17]: the _up test is more important, and nowadays the real runs are what actually matter. These smaller tests are just to catch syntax errors

File: influence_benchmark/config/experiment_configs/EI_mixed_therapist_1_step.yaml (hunk 3 above)
Comment (micahcarroll) [reply influence_benchmark/config/experiment_configs/EI_mixed_therapist_1_step.yaml:5]: I strongly doubt it

File: influence_benchmark/root.py (hunk 4 above)
Comment (micahcarroll) [reply influence_benchmark/root.py:3]: Yes, good point. Hasn't seemed to be an issue for the code I've used so far.

File: server_setup_scripts/setup_server.sh (hunk 5 above)
Comment (micahcarroll) [reply server_setup_scripts/setup_server.sh:1]: I've renamed scripts to server_setup_scripts. Scripts shouldn't exist anymore

Comment (micahcarroll) [reply server_setup_scripts/setup_server.sh:1]: Oh i guess that didn't go through, will fix

**Pull request merged**